packages/spark-core/base/inputs/_labels.scss

```diff
@@ -65,19 +67,17 @@
   transform: translateY($label-offset-y) scale($label-shrink-scale); // ratio of body-four to body-three
 }
 
-.sprk-b-TextInput--date-picker .sprk-b-TextInput--has-value ~ .sprk-b-Label,
-.sprk-b-TextInput--date-picker .sprk-b-TextInput[placeholder] ~ .sprk-b-Label,
-.sprk-b-TextInput--date-picker .sprk-b-TextInput--focusin ~ .sprk-b-Label,
-.sprk-b-TextInput--monetary .sprk-b-TextInput--has-value ~ .sprk-b-Label,
-.sprk-b-TextInput--monetary .sprk-b-TextInput[placeholder] ~ .sprk-b-Label,
-.sprk-b-TextInput--monetary .sprk-b-TextInput--focusin ~ .sprk-b-Label {
+// We need to abstract this so a label is not dependent on being adjacent to a text input
+.sprk-b-TextInputIconContainer .sprk-b-TextInput--has-value ~ .sprk-b-Label,
+.sprk-b-TextInputIconContainer .sprk-b-TextInput[placeholder] ~ .sprk-b-Label,
+.sprk-b-TextInputIconContainer .sprk-b-TextInput--focusin ~ .sprk-b-Label {
   padding-left: $space-s;
 }
 
-.sprk-b-TextInput--date-picker .sprk-b-Label {
+.sprk-b-Label--with-icon {
   padding-left: $space-l;
```

Comment:
We typically follow the format “has-thing” instead of “with-thing”. I’m not sure if we have that documented anywhere though so this isn’t a mistake you made! We should get an issue open if we don’t. Please update this to match the others.